configure.ac

```diff
@@ -807,6 +807,10 @@ if test x$use_hardening != xno; then
   case $host in
     *mingw*)
        AC_CHECK_LIB([ssp],      [main],, AC_MSG_ERROR(libssp missing))
+
+       dnl stack-clash-protection does not work on Windows
+       dnl https://gcc.gnu.org/bugzilla/show_bug.cgi?id=90458
+       AX_CHECK_COMPILE_FLAG([-fno-stack-clash-protection],[HARDENED_CXXFLAGS="$HARDENED_CXXFLAGS -fno-stack-clash-protection"])
```

Comment:
Wouldn't it be preferable in this case to not add it in the first place, instead of append `-fno-X` here? Or is it enabled by some umbrella option?

Reply:
I think it's likely we'll end up enabling this, and other flags (for other platforms) in #18921. Progress there has been slow, so I wanted to split this out and just disable it now. It doesn't work, and is causing build failures when testing 18921 and related PRs. I haven't seen it being turned on by an umbrella type option/compiler default yet.

Reply:
What I mean is making the `AC_CHECK_COMPILE_FLAG(…)` to *enable* it conditional on non-windows.

Otherwise you get a gcc overall command line like
```
fstack-clash-protection …  … fno-stack-clash-protection
```

Reply:
I agree with @wumpus. It sounds like this combo would fail to compile even the most trivial c++ program, so I can't imagine anyone shipping a mingw compiler with this on by default. So proactively turning it of seems like overkill.

Instead, how about for #18921, we add a your minimal test-case as the last (input) param to AX_CHECK_COMPILE_FLAG. That way we don't have to special-case it, and it may just work on mingw at some point in the future.

Reply:
I've merged this into #18921.